src/axom/sidre/tests/sidre_buffer.cpp

```diff
@@ -542,11 +545,24 @@ TEST_P(UmpireTest, reallocate_zero)
 const int allocators[] = {
   axom::getResourceAllocatorID( umpire::resource::Host )
 #ifdef AXOM_USE_CUDA
+
+#ifdef UMPIRE_ENABLE_PINNED
   , axom::getResourceAllocatorID( umpire::resource::Pinned )
+#endif
+
+#ifdef UMPIRE_ENABLE_DEVICE
   , axom::getResourceAllocatorID( umpire::resource::Device )
+#endif
+
+#ifdef UMPIRE_ENABLE_CONST
   , axom::getResourceAllocatorID( umpire::resource::Constant )
```

Comment:
At core/tests/core_memory_management.cpp near (new) line 400, there is a comment with some commented-out code saying "Umpire doesn't allow reallocation of Constant memory."  Some of the tests in this file and in sidre_view.cpp perform reallocation.  Will those tests need to be guarded against reallocation with a Constant memory allocator?

Reply:
Yes and the original implementation was taking that in to account by:
```
#ifdef AXOM_USE_CUDA
 if (allocID == axom::getResourceAllocatorID( umpire::resource::Constant ) )
  {
    return;
  }
#endif
```

The more precise `ifdef` guard would be:
```
#if defined(AXOM_USE_CUDA) && defined(UMPIRE_ENABLE_CONST)
  if (allocID == axom::getResourceAllocatorID( umpire::resource::Constant ) )
  {
    return;
  }
#endif
```

I made a fix. Good catch!